fuel/transformers/__init__.py

```diff
@@ -344,12 +473,26 @@ class Batch(Transformer):
         raised if a batch of the requested size cannot be provided.
 
     """
-    def __init__(self, data_stream, iteration_scheme, strictness=0):
+    def __init__(self, data_stream, iteration_scheme, strictness=0, **kwargs):
+        if not data_stream.produces_examples:
+            raise ValueError('the wrapped data stream must produce examples, '
+                             'not batches of examples.')
+        # The value for `produces_examples` is inferred from the iteration
+        # scheme's `requests_examples` attribute. We expect the scheme to
+        # request batches.
+        if iteration_scheme.requests_examples:
+            raise ValueError('the iteration scheme must request batches, '
+                             'not individual examples.')
+        if data_stream.axis_labels:
+            kwargs.setdefault(
```

Comment:
Same question about whether this should be overrideable. I guess it's a question of flexibility versus sanity checking. Screw it, let's leave it as is.